keanu-project/src/main/java/io/improbable/keanu/tensor/Tensor.java

```diff
@@ -31,13 +31,17 @@ static BooleanTensor elementwiseEquals(Tensor a, Tensor b) {
     }
 
     static <T> Tensor<T> scalar(T value) {
-        return new GenericTensor<>(value);
+        return GenericTensor.scalar(value);
     }
 
     static <T> Tensor<T> placeHolder(long[] shape) {
         return new GenericTensor<>(shape);
     }
 
+    static <T> GenericTensor<T> create(T value, long[] shape) {
```

Comment:
We should also add specific versions , i.e.

    static DoubleTensor create(Double value, long[] shape) {
        return DoubleTensor.create(value, shape);
    }

    static BooleanTensor create(Boolean value, long[] shape) {
        return BooleanTensor.create(value, shape);
    }

    static IntegerTensor create(Integer value, long[] shape) {
        return IntegerTensor.create(value, shape);
    }